extensions/merge-conflict/src/contentProvider.ts

```diff
@@ -23,12 +23,40 @@ export default class MergeConflictContentProvider implements vscode.TextDocument
 
 	async provideTextDocumentContent(uri: vscode.Uri): Promise<string | null> {
 		try {
-			const { scheme, range } = JSON.parse(uri.query) as { scheme: string; range: { line: number, character: number }[] };
-			const [start, end] = range;
+			const { type, scheme, range, fullRange, ranges } = JSON.parse(uri.query) as { type: string, scheme: string; range: { line: number, character: number }[], fullRange: { line: number, character: number }[], ranges: [{ line: number, character: number }[], { line: number, character: number }[]][] };
 
-			const document = await vscode.workspace.openTextDocument(uri.with({ scheme, query: '' }));
-			const text = document.getText(new vscode.Range(start.line, start.character, end.line, end.character));
-			return text;
+			if (type === 'full') {
+				// complete diff
+				const document = await vscode.workspace.openTextDocument(uri.with({ scheme, query: '' }));
+
+				let text = '';
+				let lastPosition = new vscode.Position(0, 0);
+				ranges.forEach(rangeObj => {
```

Comment:
We concatenate the before/after ranges to two new documents and then diff the two - if I understand correctly. This could result in a different diff than when comparing the two original before/after documents. Is that going to be a problem?

Reply:
> This could result in a different diff than when comparing the two original before/after documents 

Yes and no, let me elaborate more. *No*, because the `after` document is the file content on disk, which already resolves some conflicts successfully, so `compareAll` shows the difference of conflicts which are not resolved. From this perspective, the diff view of `compareAll` is correct.

Yes if we are thinking in a three-diff way. Say the users is trying to merge two branch, whose heads are A and B. A is upstream, and B is a branch you created a few weeks ago. After merging, there is a merge conflict. The diff view of `compareAll` is different from `A --- B` then. However in this case, a simple comparison between `A` and `B` doesn't really work as most changes in `A --- B` are probably not related to the work you are doing. To avoid confusion, you may want to see a three-way-diff editor, which represents the diff view between A, B and their common nearest ancestor. We can do this later on as it requires introducing a new diff algorithm and editor pane.